msteams-platform/sbs-botbuilder-conversation-AI.yml

```diff
@@ -0,0 +1,709 @@
+            * The `MemoryStorage` class is imported from the Bot Builder SDK and has a predefined logic to handle any issues when two or more events exist at the same time so that you don't want to worry about handling storage issues and focus on building your application.
+
+
+            1. Under **Import required bot service**, add `MemoryStorage`.
+
+        1.  Add the following code to define actions:
+
+            ```typescript
+            app.ai.action(AI.FlaggedInputActionName, async (context, state, data) => {
+                await context.sendActivity(`I'm sorry your message was flagged: ${JSON.stringify(data)}`);
+                return false;
+            });
+
+            app.ai.action(AI.FlaggedOutputActionName, async (context, state, data) => {
+                await context.sendActivity(`I'm not allowed to talk about such things.`);
+                return false;
+            });
+            ```
+            
+            When the `OpenAIModerator` reviews and identifies a flag in an input or output, it triggers the action `AI.FlaggedInputActionName` or `AI.FlaggedOutputActionName` and the `context.sendActivity` sends a message to the user. If the user receives a message for an flagged  input or output, the user can control what he wants to do.
+
+        1. Add the following code to define history:
+
+            ```typescript
+            app.message('/history', async (context, state) => {
+            const history = ConversationHistory.toString(state, 2000, '\n\n');
+            await context.sendActivity(history);
+            });
+            ```
+
+           The `ConversationHistory` class remembers context across messages. It automatically populates the {{$history}} variable available in the `prompt.txt` file.
+
+        1. By default Bot builder uses `bot` as a variable, since we are constructing an app for the `new Application`. We need to replace `await bot.run(context) with `await app.run(context)`.
+   
+    1. In the root folder, create a new folder **prompts/chat**.
+
+    1. Under **chat**, create the following files:
+       * config.json
+       * skprompt.txt
+
+    1. Go to the `config.json` file, and add the following code to create the prompt model settings:
+    
+        ```javascript
+        {
+            "schema": 1,
+            "description": "Chat with Teams Chef",
+            "type": "completion",
+            "completion": {
+            "max_tokens": 150,
+            "temperature": 0.9,
+            "top_p": 0.0,
+            "presence_penalty": 0.6,
+            "frequency_penalty": 0.0,
+            "stop_sequences": [
+                "Human:",
+                "AI:"
+            ]
+            }
+        }
+        ```
+
+        The `config.json' is initiated by the `DefaultPromptManager` from the `index.ts` file. When a prompt is initiated, the prompt manager searches for  the prompt config file (chat/config.json) and passed the prompt settings into Open AI or Azure Open AI.
```

Comment:
The tick (`) is messed up on the "config.json" text:

<img width="317" alt="image" src="https://github.com/MicrosoftDocs/msteams-docs/assets/115390646/85420b4b-4d00-446e-9552-ed693cf7c2a6">

Reply:
updated.